crypto/src/proofs/transparent.rs

```diff
@@ -557,97 +472,49 @@ impl TryFrom<transparent_proofs::SwapClaimProof> for SwapClaimProof {
-
-        let swap_plaintext = SwapPlaintext {
-            trading_pair,
-            delta_1_i: self.value_t1.amount,
-            delta_2_i: self.value_t2.amount,
-            claim_fee: self.fee_delta.clone(),
-            claim_address: self.claim_address,
-            swap_blinding: self.swap_blinding,
-        };
-
-        // Checks the note commitment of the Swap NFT.
-        gadgets::note_commitment_integrity(
-            Note::from_parts(
-                self.claim_address,
-                Value {
-                    // The swap NFT is always amount 1.
-                    amount: 1u64.into(),
-                    asset_id: asset::Id(swap_plaintext.swap_commitment().0),
-                },
-                self.note_blinding,
-            )?,
-            note_commitment,
-        )?;
-
-        // TODO: no value commitment checks until flow encryption is available
-        // // Value commitment integrity.
-        // if value_1_commitment != -self.value_t1.commit(self.delta_1_blinding) {
-        //     return Err(anyhow!("value commitment mismatch"));
-        // }
-
-        // if value_2_commitment != -self.value_t2.commit(self.delta_2_blinding) {
-        //     return Err(anyhow!("value commitment mismatch"));
-        // }
-
-        gadgets::balance_commitment_integrity(
-            value_fee_commitment,
-            self.fee_blinding,
-            Balance::from(self.fee_delta.0),
-        )?;
+        // Swap commitment integrity check
+        ensure!(
+            swap_commitment == self.swap_plaintext.swap_commitment(),
+            "swap commitment mismatch"
+        );
 
-        gadgets::ephemeral_public_key_integrity(
-            epk,
-            self.esk.clone(),
-            self.claim_address.diversified_generator().clone(),
-        )?;
+        // Fee commitment integrity check
+        ensure!(
+            fee_commitment == self.swap_plaintext.claim_fee.commit(self.fee_blinding),
+            "fee commitment mismatch"
+        );
 
-        gadgets::diversified_basepoint_not_identity(
-            self.claim_address.diversified_generator().clone(),
-        )?;
+        // Now reconstruct the swap action's balance commitment
+        let transparent_balance = Balance::default()
+            - Value {
+                amount: self.swap_plaintext.delta_1_i,
+                asset_id: self.swap_plaintext.trading_pair.asset_1(),
+            }
+            - Value {
+                amount: self.swap_plaintext.delta_2_i,
+                asset_id: self.swap_plaintext.trading_pair.asset_2(),
+            };
+        let transparent_balance_commitment = transparent_balance.commit(Fr::zero());
+
+        // XXX sign error here
+        // XXX we want to avoid having to twiddle signs for synthetic blinding factor in binding sig
+        ensure!(
+            balance_commitment == transparent_balance_commitment + fee_commitment,
+            "balance commitment mismatch"
+        );
```

Comment:
Followup: review signs for fee commitment in Swap actions

I don't think either this code or the original code was correctly handling signs for the fee commitment. I suspect there are bugs lurking that we only miss because we set 0 fees by default.

We should do a followup pass and:
- [ ] review the signs of fees as they're used in `Swap`s
- [ ] write a state machine integration test that creates and executes complete, fully-valid `Transaction`s (using the test account's note state) that sets distinct nonzero fees for both `Swap` and `SwapClaim` transactions.